debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresHeartbeat.java

```diff
@@ -0,0 +1,64 @@
+package io.debezium.connector.postgresql;
+
+import java.util.Map;
+
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.connect.source.SourceRecord;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.debezium.config.Configuration;
+import io.debezium.config.Field;
+import io.debezium.connector.postgresql.connection.PostgresConnection;
+import io.debezium.function.BlockingConsumer;
+import io.debezium.heartbeat.HeartbeatImpl;
+
+/**
+ * Postgres specific implementaiton of the heartbeat feature that allows for a PG query to be executed with every heartbeat.
+ */
+public class PostgresHeartbeat extends HeartbeatImpl {
+    private static final Logger LOGGER = LoggerFactory.getLogger(PostgresHeartbeat.class);
+
+    public static final String HEARTBEAT_ACTION_ENABLED_PROPERTY_NAME = "heartbeat.action.enabled";
+    public static final String HEARTBEAT_ACTION_QUERY_PROPERTY_NAME = "heartbeat.action.query";
+
+    public static final Field HEARTBEAT_ACTION_ENABLED = Field.create(HEARTBEAT_ACTION_ENABLED_PROPERTY_NAME)
+            .withDisplayName("Enables the heartbeat action")
+            .withType(ConfigDef.Type.BOOLEAN)
+            .withImportance(ConfigDef.Importance.LOW)
+            .withDescription("Enables the heartbeat.action.query to be executed with every heartbeat. Defaults to false.")
+            .withDefault(false);
+
+    public static final Field HEARTBEAT_ACTION_QUERY = Field.create(HEARTBEAT_ACTION_QUERY_PROPERTY_NAME)
+            .withDisplayName("The query to execute with every heartbeat")
+            .withType(ConfigDef.Type.STRING)
+            .withWidth(ConfigDef.Width.MEDIUM)
+            .withImportance(ConfigDef.Importance.LOW)
+            .withDescription("The query executed with every heartbeat. Defaults to an empty string.")
+            .withDefault("");
+
+    private final boolean heartBeatActionEnabled;
+    private final String heartBeatActionQuery;
+    private final PostgresConnection postgresConnection;
+
+    PostgresHeartbeat(Configuration configuration, String topicName, String key, PostgresConnection postgresConnection) {
```

Comment:
I'd not call it postgresheartbeat but databaseheartbeat and move it into core module. JdbcConnection can be used as parameter.

Reply:
Good idea, done.